### ISSUE TITLE
WithProductionDoorOverlay crashes if produced unit isn't Positionable.

### ISSUE THREAD

**Issue by MustaphaTR** [OpenRA/OpenRA#20772]
Noticed while updating Generals Alpha to release-20200503, game crashes when building War Factory upgrades. Haven't tested on bleed yet, but the code around that doesn't seem to have changed since then. The door animation should just ignore such actors and not play.

**Comment by ThomasChr** [comment]
Hi @MustaphaTR and thanks for your bug report!
Could you elaborate a little bit further? tbh, I don't know anything about Generals or War Factory upgrades.

**Comment by anvilvapre** [comment]
I don't see any direct references from `WithProductionDoorOverlay` to `IPositionable`. 

Please provide stack traces along with crash reports. And/or a clearer explenation of the issue.

**Comment by MustaphaTR** [comment]
Has been a while, but `existingActor.Location` [here](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/Render/WithProductionDoorOverlay.cs#L64) should be the issue. It is null for the dummy upgrade actors.

**Comment by anvilvapre** [comment]
the line above checks to see if it is null?

**Comment by MustaphaTR** [comment]
Location, not the actor itself.

**Comment by anvilvapre** [comment]
`Actor.cs` `public CPos Location => OccupiesSpace.TopLeft;` OccupiesSpace is IPositionable, which must be null then. The trait lacking.

**Comment by MustaphaTR** [comment]
Yeah, it is not supposed to Occupy Space, it is an upgrade dummy. Which is why this should be fixed on WPDO's end. I already have the check on GenAlpha's engine. I can do a PR upstream if i can find time.
